crates/rpc-types-eth/src/filter.rs

```diff
@@ -193,6 +193,46 @@ impl FilterBlockOption {
             Self::AtBlockHash(_) => (None, None),
         }
     }
+
+    /// Returns the block hash if this is a block hash filter.
+    pub const fn as_block_hash(&self) -> Option<&BlockHash> {
+        match self {
+            Self::AtBlockHash(hash) => Some(hash),
+            Self::Range { .. } => None,
+        }
+    }
+
+    /// Returns true if this is a range filter.
+    pub const fn is_range(&self) -> bool {
+        matches!(self, Self::Range { .. })
+    }
+
+    /// Returns true if this is a block hash filter.
+    pub const fn is_block_hash(&self) -> bool {
+        matches!(self, Self::AtBlockHash(_))
+    }
+
+    /// Sets the block number this range filter should start at.
+    #[must_use]
+    pub const fn set_from_block(&self, block: BlockNumberOrTag) -> Self {
```

Comment:
set_ is wrong naming because it takes by reference

Reply:
right, perhaps we can either drop the prefix or replace with `with_`

Reply:
Initially I just wanted to move this implementation that was already done here to reduce to a single `impl FilterBlockOption` instead of two. I hope this is ok to change the API :)